usmqe/gluster/gluster.py

```diff
@@ -202,7 +189,39 @@ def info(self):
 
         Returns:
             dictionary: parsed json from
-                        ``gluster --format json --cluster CLUSTERNAME status``
+                        ``gluster volume info VOLUMENAME --xml``
                         command
         """
-        return self.run_on_node('info')
+        xml = self.run_on_node('info {}'.format(self.name))
+        self.id = xml.findtext("./volInfo/volumes/volume/id")
+        LOGGER.debug("Volume_id: %s" % self.id)
+        self.status = xml.findtext(
+            "./volInfo/volumes/volume/statusStr")
+        self.stripe_count = xml.findtext(
+            "./volInfo/volumes/volume/stripeCount")
+        self.replica_count = xml.findtext(
+            "./volInfo/volumes/volume/stripeCount")
+        self.brick_count = xml.findtext(
+            "./volInfo/volumes/volume/brickCount")
+        self.snap_count = xml.findtext(
+            "./volInfo/volumes/volume/snapshotCount")
+        LOGGER.debug("Volume_status: %s" % self.status)
+
+    def get_volume_id(self):
```

Comment:
In `get_volume_id` method is `info` called only if id is undefined in object and id is defined in object only in case that `info` was called.
In `check_status` method I call `info` every time because in case of status you always want current status I think.